### PR TITLE
feat(pages): /page slash command creates and links a sub-page

### PR DESCRIPTION
**Pull request by positonic**
## What

Typing `/` in a Knowledge Page body now offers a **Page** command (filterable as `/page`). Selecting it:

1. creates a new page in the same workspace/project (inheriting identical visibility),
2. inserts a `pageLink` block at the cursor,
3. flushes the autosave (so the link survives the navigation unmount), and
4. opens the new page.

Navigating back shows the link with a file icon and the target page's title.

## How the title stays live

The `pageLink` node stores only the target `pageId` plus a cached `title`/`href` snapshot. A React node view resolves the display title from the shared `page.list` query — the same query `PageTitle` already invalidates on rename — so one cached query serves every link in a document and renames propagate to all links without rewriting the linking documents. While a doc is editable, the live title is synced back into the node attr so the Markdown projection and public render stay current.

## Off-island + public behavior

- **Markdown projection** (`body`, read by CLI/agents): the node serialises as a plain `[title](/w/{slug}/pages/{id})` link. There is no parse rule — a Markdown-source rewrite (e.g. Zoe) degrades it to a regular link mark, which still navigates in-app.
- **Published pages** (ADR-0038): `sanitizeDocForPublic` strips `pageId` and the internal workspace path from page links; the public HTML shows just the title text, so internal URLs/ids never leak.
- The node lives in the shared PRD schema (`buildPrdExtensions`), with an override point so the live editor attaches the React view while the headless codec and server-side `generateHTML` use the plain base node (keeps the server bundle free of client imports).

## Changes

- `src/lib/prd/page-link.ts` (new) — base `pageLink` block node: schema, static HTML render, Markdown serialisation
- `src/lib/prd/slash-command.tsx` — slash menu accepts host-injected `extraCommands`
- `src/lib/prd/extensions.ts` — registers the node; optional node-view override
- `src/app/_components/shared/PageLinkView.tsx` (new) — interactive node view with live title + navigation
- `src/app/_components/shared/RichDocEditor.tsx` — `slashExtras` prop; uses the view-enabled node
- `src/app/_components/pages/PageDocument.tsx` — the `/page` command: create → insert → flush save → invalidate caches → navigate
- `src/app/(sidemenu)/w/[workspaceSlug]/pages/[id]/page.tsx` — passes workspace/project placement down
- `src/lib/pages/public-doc.ts` — public sanitizer handling for `pageLink`

## Tests

- `codec.test.ts`: Markdown projection of `pageLink` (happy path, bracket escaping, missing attrs)
- `public-doc.test.ts`: public sanitization strips id/href, keeps title
- Full unit suite green (1251 tests); typecheck and ESLint clean

## Notes

- New pages are titled **"Untitled"**, matching the existing "New page" button.
- No DB changes — safe for main per the fast-track criteria.

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Adds `/page` slash command to Knowledge Page editor creating linked sub-pages

- Introduces new `pageLink` TipTap node with Markdown serialization and public sanitization

- `flushSave` now returns a `Promise<void>` ensuring save completes before navigation

- Adds `SlashCommandItem` export and `extraCommands` option for host-injected commands

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["PageDocument\n(host)"] -- "slashExtras + onReady" --> B["RichDocEditor"]
  B -- "buildPrdExtensions\n(pageLink override)" --> C["PageLinkWithView\n(React node view)"]
  C -- "page.list query" --> D["Live title resolution"]
  B -- "SlashCommand.configure\n(extraCommands)" --> E["SlashCommand extension"]
  E -- "/page selected" --> F["createPage mutation"]
  F -- "insertContent pageLink" --> G["pageLink node"]
  G -- "flushSave (awaited)" --> H["Navigate to new page"]
  G -- "Markdown serialization" --> I["[title](href)"]
  G -- "sanitizeDocForPublic" --> J["title only (no pageId/href)"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td><strong>page-link.ts</strong><dd><code>New pageLink block node with Markdown serialization</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-94119179092d410fa0aa1596920ee27e263807f9ede48067f1c6fa61b4a1719a">+92/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>PageLinkView.tsx</strong><dd><code>React node view resolving live page title</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-a4f054452bc17097d18c6cb103a6d0413d6408615cf8ed187d32967ec0295398">+72/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>PageDocument.tsx</strong><dd><code>Wire /page slash command and flushSave handle</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-381f79c00467608f97330626d377b83d5ae351da580b4673a46edc2af783f1af">+82/-1</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>page.tsx</strong><dd><code>Pass workspaceId, workspaceSlug, projectId to PageDocument</code></dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-6b79b72a995acbd10cc157f1131181cd7ed46fc478f1fd996884e2c162db08b2">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>RichDocEditor.tsx</strong><dd><code>flushSave returns Promise; add slashExtras and PageLinkWithView</code></dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-94ac9cac79aa08ce267f273a39bca6d43488b86f079813d3132a5a8770c18913">+24/-11</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>extensions.ts</strong><dd><code>Add pageLink node with override option to buildPrdExtensions</code></dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-df9e8e0d34ac894a019436e84fc03a982819ea60173d72157688eb406a309dc8">+13/-2</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>slash-command.tsx</strong><dd><code>Export SlashCommandItem and add extraCommands option</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-acdef4f8d766b21f181571a45e27e07b8527191f38b36f90e682ea9c15246a6e">+24/-9</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>public-doc.ts</strong><dd><code>Strip pageId and internal href from pageLink in public render</code></dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-64d759dc1f727883c8edd8f63a004f8f1f145b831eeb35e116d75a036cf4e0d4">+10/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>PrdDocument.tsx</strong><dd><code>Update flushSaveRef type to Promise-returning</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-fb26fc25e3f6039b3fcca439dad29a1ebe61368c3ea825b772515b3acae5dd35">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>public-doc.test.ts</strong><dd><code>Tests for pageLink sanitization in public doc</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-17f4aa034ebdbb5efd00ae1ff6e130fe9048ad7bb627a89d746270d47dda885c">+23/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>codec.test.ts</strong><dd><code>Tests for pageLink Markdown serialization and edge cases</code>&nbsp; </dd></td>
  <td><a href="https://github.com/positonic/exponential/pull/343/files#diff-2ba7b1a5c15e6ebf19a9cb4982363f8a5e224c3c6f44bbc3c6caf5d4bfd7e690">+44/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: src/app/_components/pages/PageDocument.tsx [hunk 1]
```diff
@@ -12,24 +20,90 @@ interface PageDocumentProps {
...
+                  })
+                  .run();
+                // Persist the link now — navigating away unmounts the editor
+                // before the debounced autosave would fire — and mark this
+                // page's cached doc stale so back-navigation refetches it.
+                handleRef.current?.flushSave();
+                void utils.page.get.invalidate({ id: pageId });
+                void utils.page.list.invalidate();
+                router.push(`/w/${workspaceSlug}/pages/${page.id}`);
+              },
```
[the remark this answers is not in the document]
Comment (positonic) [reply src/app/_components/pages/PageDocument.tsx]: ✅ Addressed in [`720eaa9f`](https://github.com/positonic/exponential/pull/343/commits/720eaa9f) — fix(pages): await the flushed save before navigating to a new sub-page

File: src/lib/prd/page-link.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    return {
+      markdown: {
+        serialize(state: MarkdownSerializerState, node: ProseMirrorNode) {
+          const title =
+            ((node.attrs.title as string | null) ?? "Untitled") || "Untitled";
+          const href = (node.attrs.href as string | null) ?? "";
+          const safeTitle = title.replace(/([[\]])/g, "\\$1");
+          state.write(`[${safeTitle}](${href})`);
+          state.closeBlock(node);
+        },
```
[the remark this answers is not in the document]
Comment (positonic) [reply src/lib/prd/page-link.ts:87]: ✅ Addressed in [`7e2b5baf`](https://github.com/positonic/exponential/pull/343/commits/7e2b5baf) — fix(pages): escape Markdown-sensitive characters in pageLink hrefs
